rule-engine/rule-engine-components/src/main/java/org/thingsboard/rule/engine/metadata/CalculateDeltaNode.java

```diff
@@ -57,19 +58,24 @@
 public class CalculateDeltaNode implements TbNode {
 
     private Map<EntityId, ValueWithTs> cache;
+    private Map<EntityId, SemaphoreWithTbMsgQueue> locks;
```

Comment:
the idea of synchronization by entityId is that the rule nodes can be used multiple times across all rule chains and the single entity will run in concurrent mode for this node.
Until you use ` private static final` for locks and cache. Seen TbMathNode

Reply:
Confirmed internally that synchronization over all instances of this node is not needed in case of this node